services/galley/test/integration/API/CustomBackend.hs

```diff
@@ -52,9 +51,11 @@ getByDomainInvalidDomain :: TestM ()
 getByDomainInvalidDomain = do
   galley <- view tsGalley
   -- contains invalid character '+'
+  -- this used to respond with '400 bad request'
+  -- but after servantification it returns '404 not found'
+  -- because the domain parameter is invalid and therefore the endpoint is invalid, too
   get (galley . path "/custom-backend/by-domain/invalid%2Bdomain") !!! do
-    const 400 === statusCode
-    const ("client-error" :: ByteString) =~= (cs . fold . responseBody)
+    const 404 === statusCode
```

Comment:
I can't see what in this PR changes the status code. From what I can see, when an endpoint throws `'CustomBackendNotFound`, this has always mapped to a 404 as it can be seen in `libs/wire-api/src/Wire/API/Error/Galley.hs`:

```haskell
type instance MapError 'CustomBackendNotFound = 'StaticError 404 "custom-backend-not-found" "Custom backend not found"
```

Can you point me to the place where your PR changes the status code?

Reply:
that is a good question, my assumption was that in the wai-routes definition [here](https://github.com/wireapp/wire-server/pull/2286/files#diff-f577bd929063ed2e5a0fc6ded4e6ecce3948249ef3812dfddcf77c3db91b6398L394):

```
  get "/custom-backend/by-domain/:domain" (continueE CustomBackend.getCustomBackendByDomainH) $
    capture "domain"
```
`domain` is treated as user input and therefore the response is 400
whereas servant treats it as a path and if it is not parsable the response is 404

Reply:
Oh, OK, so it's how Servant handles the issue vs. how wai-routes handles it.